src/NuGet.Core/NuGet.Configuration/PackageNamespaces/PackageNamespacesConfiguration.cs

```diff
@@ -38,11 +50,19 @@ public IReadOnlyList<string> GetConfiguredPackageSources(string term)
             return SearchTree.Value?.GetConfiguredPackageSources(term);
         }
 
-        internal PackageNamespacesConfiguration(Dictionary<string, IReadOnlyList<string>> namespaces)
+        internal PackageNamespacesConfiguration(Dictionary<string, IReadOnlyList<string>> namespaces, PackageNamespacesProvider packageNamespacesProvider)
```

Comment:
I think it's unnecessary to pass down a PackageNamespaceProvider here because we're only using it for the new `GetPackageSources` method. 
It'd be cleaner to provide the list of package sources instead.

Reply:
I'd also recommend a null check to have the correct expectations about the API.